fprime-zephyr/Drv/LoRa/LoRa.cpp

```diff
@@ -109,29 +114,33 @@ void LoRa ::dataIn_handler(FwIndexType portNum, Fw::Buffer& data, const ComCfg::
     FW_ASSERT(data.getSize() + sizeof(LoRaConfig::HEADER) <= LoRa::MAX_PACKET_SIZE, data.getSize());
     FW_ASSERT(this->m_lora_device != nullptr);
     FW_ASSERT(device_is_ready(this->m_lora_device));
-    Fw::Success returnStatus = Fw::Success::SUCCESS;
-    Status status = this->enableTx();
-    if (status == Status::SUCCESS) {
-        (void)::memcpy(this->m_send_buffer, LoRaConfig::HEADER, sizeof(LoRaConfig::HEADER));
-        (void)::memcpy(this->m_send_buffer + sizeof(LoRaConfig::HEADER), data.getData(), data.getSize());
-        int send_status =
-            lora_send(this->m_lora_device, this->m_send_buffer, sizeof(LoRaConfig::HEADER) + data.getSize());
-        if (send_status != 0) {
-            this->log_WARNING_HI_SendFailed(static_cast<I32>(send_status));
-            returnStatus = Fw::Success::FAILURE;
+    Fw::Success returnStatus = Fw::Success::FAILURE;
+    if (this->m_transmit_enabled == TransmitState::ENABLED) {
+        Status status = this->enableTx();
+        if (status == Status::SUCCESS) {
+            (void)::memcpy(this->m_send_buffer, LoRaConfig::HEADER, sizeof(LoRaConfig::HEADER));
+            (void)::memcpy(this->m_send_buffer + sizeof(LoRaConfig::HEADER), data.getData(), data.getSize());
+            int send_status =
+                lora_send(this->m_lora_device, this->m_send_buffer, sizeof(LoRaConfig::HEADER) + data.getSize());
+            if (send_status != 0) {
+                this->log_WARNING_HI_SendFailed(static_cast<I32>(send_status));
+                returnStatus = Fw::Success::FAILURE;
+            } else {
+                returnStatus = Fw::Success::SUCCESS;
+                this->log_WARNING_HI_ConfigurationFailed_ThrottleClear();
+                this->log_WARNING_HI_SendFailed_ThrottleClear();
+            }
         } else {
-            returnStatus = Fw::Success::SUCCESS;
-            this->log_WARNING_HI_ConfigurationFailed_ThrottleClear();
-            this->log_WARNING_HI_SendFailed_ThrottleClear();
+            this->log_WARNING_HI_ConfigurationFailed(LoRaMode::Transmit);
+            returnStatus = Fw::Success::FAILURE;
         }
-    } else {
-        this->log_WARNING_HI_ConfigurationFailed(LoRaMode::Transmit);
-        returnStatus = Fw::Success::FAILURE;
-    }
-    // Enable RX mode after all TXes
-    status = this->enableRx();
-    if (status != Status::SUCCESS) {
-        this->log_WARNING_HI_ConfigurationFailed(LoRaMode::Receive);
+        // Enable RX mode after all TXes
+        status = this->enableRx();
+        if (status != Status::SUCCESS) {
+            this->log_WARNING_HI_ConfigurationFailed(LoRaMode::Receive);
+        }
+    } else if (this->m_transmit_enabled == TransmitState::DISABLING) {
+        this->m_transmit_enabled = TransmitState::DISABLED;
```

Comment:
Do you need to set the returnStatus to success here? Otherwise, you'll call comStatusOut with returnStatus' default value (which is a FAILURE).